### ISSUE TITLE
Dúvida Geral

### ISSUE THREAD

**Issue by maik001** [nfephp-org/sped-esocial#408]
Estou começando no sped-esocial agora e gostaria de saber se existem, e se existirem onde se localizam os seguintes módulos:
- Envio de Eventos para o Ambiente de Teste do eSocial
- Monitoramento dos Lotes de Eventos em Fila
- Monitoramento se o Evento já foi processado pelo o eSocial e se foi, qual o Retorno.
Se possível enviar exemplos.

**Comment by andersonls** [comment]
Dê uma olhada nos arquivos de exemplo:

- [examples](https://github.com/nfephp-org/sped-esocial/tree/master/examples)/[Fake](https://github.com/nfephp-org/sped-esocial/tree/master/examples/Fake)/[v_S_01_00_00](https://github.com/nfephp-org/sped-esocial/tree/master/examples/Fake/v_S_01_00_00)/FaketestEnvia.php
- [examples](https://github.com/nfephp-org/sped-esocial/tree/master/examples)/[Fake](https://github.com/nfephp-org/sped-esocial/tree/master/examples/Fake)/[v_S_01_00_00](https://github.com/nfephp-org/sped-esocial/tree/master/examples/Fake/v_S_01_00_00)/FaketestConsultaLote.php

Estes são os dois arquivos de exemplo para envio e consulta de lote de eventos.

**Comment by maik001** [comment]
Mas, por exemplo, o FakeTestEnvia me retorna um Soap pelo que eu acabei de testar aqui com os dados da requisição que eu tô mandado, mas onde eu pego o Recibo que é necessário pra fazer consulta do lote do evento?

**Comment by andersonls** [comment]
Os métodos que enviam dados para o E-social sempre retornam o próprio XML retornado pelo E-social, você vai precisar processar estes XMLs (no manual do E-social você pode ver o layout).

Por exemplo, para pegar o protocolo de envio eu faço o seguinte (que usa na consulta do lote):

```
        $response = $tools->enviarLoteEventos($tools::EVT_INICIAIS, [$evento]);

        $dom = new \DOMDocument();
        $dom->loadXML($response);

        $protocoloEnvio = $dom->getElementsByTagName('protocoloEnvio')->item(0)->nodeValue;
```

O recibo só é retornado na consulta do lote, depois que o evento foi processado e aceito pelo E-social.

**Comment by maik001** [comment]
Exato, mas como eu sei se o eSocial aprovou ou não o evento, no projeto do TSTLabs por exemplo tem um endpoint chamado ocorrencias onde ele tem um json com o retorno do eSocial.

**Comment by maik001** [comment]
Beleza entendi, Obrigado pela Ajuda.
